src/test/java/io/fabric8/maven/docker/MojoTestBase.java

```diff
@@ -251,6 +251,7 @@ protected void givenMavenProject(AbstractDockerMojo mojo) {
         mojo.project = mavenProject;
         mojo.log = ansiLogger;
         mojo.outputDirectory= "target/docker";
+        mojo.sourceDirectory = "src/main/docker";
```

Comment:
I added this to prevent a NullPointerException in the test [buildUsingConfiguredBuildxWithContext](https://github.com/fabric8io/docker-maven-plugin/blob/master/src/test/java/io/fabric8/maven/docker/BuildMojoTest.java#L116). It sets a contextDir on the BuildImageConfiguration, which causes a default dockerFileFile to be set. This causes the BuildService to attempt to get the FROM images from the Dockerfile, and the source directory needs to be set to calculate the absolute path to the file.